gwpy/cli/histogram.py

```diff
@@ -0,0 +1,187 @@
+        self.ymin = sys.maxsize
+        self.ymax = 0
+
+    @classmethod
+    def init_cli(self, parser):
+        """We add an option to add a fit ot gaussian PDF"""
+        super().init_cli(parser) # add all timeseries options
+        group = parser.add_argument_group('Curve fit options')
+        group.add_argument('--gauss', action='store_true',
+                           help='Add gaussian fit')
+
+    @classmethod
+    def get_xlabel(self):
+        """Text for x-axis label
+        """
+        units = self.units
+        if len(units) == 1 and str(units[0]) == '':  # dimensionless
+            return 'Sample value'
+        if len(units) == 1 and self.usetex:
+            return 'Sample value ({})'.format(units[0].to_string('latex'))
+        elif len(units) == 1:
+            return 'Sample value ({})'.format(units[0].to_string())
+        elif len(units) > 1:
+            return 'Sample value (multiple units)'
+        return 'Sample value'
+
+    @classmethod
+    def get_ylabel(self):
+        """Text for y-axis label
+        """
+        return 'Count'
+
+    @classmethod
+    def get_suptitle(self):
+        """Start of default super title, first channel is appended to it
+        """
+        return 'Histogram: {0}'.format(self.chan_list[0])
+
+    @classmethod
+    def get_title(self):
+        suffix = super().get_title()
+        # limit significant digits for minute trends
+        rates = {ts.sample_rate.round(3) for ts in self.timeseries}
+        fss = '({0})'.format('), ('.join(map(str, rates)))
+        ret = ', '.join([
+            'Fs: {0}'.format(fss),
+            'dur: {0}'.format(self.duration),
+            suffix,
+        ])
+        ret = '{} bins: {}, mean: {:.3g}, sd: {:.3g}  ' \
+              'skew: {:.3g}, kurtosis: {:.3g}'.\
+            format(ret, self.nbins, self.mean.value, self.std.value,
+                   self.skew, self.kurtosis)
+        return ret
+
+    @classmethod
+    def make_plot(self):
+        """Generate the plot from time series and arguments
+        """
+        plot = Plot(figsize=self.figsize, dpi=self.dpi)
+        ax = plot.gca(xscale='linear')
+
+        # handle user specified plot labels
+        if self.args.legend:
+            nlegargs = len(self.args.legend[0])
+        else:
+            nlegargs = 0
+        if nlegargs > 0 and nlegargs != self.n_datasets:
+            warnings.warn('The number of legends specified must match '
+                          'the number of time series'
+                          ' (channels * start times).  '
+                          'There are {:d} series and {:d} legends'.format(
+                            len(self.timeseries), len(self.args.legend)))
+            nlegargs = 0    # don't use  them
+
+        for i in range(0, self.n_datasets):
+            series = self.timeseries[i]
+            if nlegargs:
+                label = self.args.legend[0][i]
+            else:
+                label = series.channel.name
```

Comment:
This can be done on one line:

```suggestion
            label = self.args.legend[0][i] if nlegargs else series.channel.name
```

Reply:
These both now use the conditional assignment